### DISCUSSION TITLE
updating user password causes subsequent updates to fail

### DISCUSSION THREAD

**Discussion by archoo** [pocketbase/pocketbase#3281] [Q&A] [answered]
I'm trying to build a simple user profile single page app, and have most things functioning.. I can create users, login/out and update details including custom fields in the users collection.. this all works nicely.. 

however whenever I update the users password, any subsequent calls to update() get presented with a 404 response..  the password *is* changed successfully, and refreshing the page and logging in again with the new password works ok.. 

what is confusing me is that i can call pb.authStore.isValid in the console immediately after the http call that performs the password change and it returns 'true' .. but calling authRefresh in the console immediately afterward gets a 401 error?

![image](https://github.com/pocketbase/pocketbase/assets/133547/bf34c32f-6fb5-4839-a862-afee6a47cd15)
(edit: i've confirmed the authorization header is set correctly in that auth-refresh http call)

i've trawled through old posts here and i understand that the api rules return 404 if they fail, so i'm pretty sure the user is no longer considered valid to perform updates after the password change?

I tried clearing the authStore on password update and forcing the user to login with the new credentials straight away, which works, but i'd rather not do that since its jarring from a user flow point of view.. 

so it appears that doing a password update invalidates the client token, but the response from the update() api endpoint doesn't return an updated token and the suggested method to refresh it (ie call authRefresh) then fails so i'm a little lost on what steps to take?  i'd like to be able to update my password and stay authorised? 

what am i missing??

**Comment by ganigeorgiev** [comment, marked as answer]
This is not documented very well at the moment, I'll consider adding a note in the API preview.

When changing passwords all previously issued tokens are automatically invalidated, meaning that you need to reauthenticate manually after the password change request.

_Note that `pb.authStore` only loosely checks whether the stored token is valid or not (aka. if it is a proper JWT with non-expired `exp` claim), but doesn't perform checks against the PocketBase server. Token validation checks are performed only when making requests to the PocketBase server._

**Reply by einarpersson** [reply]
We just stumbled on this as well. Would be great to add a note in the docs / API preview! :)

**Reply by ganigeorgiev** [reply]
There should be already a note in the action description in the API preview docs, but with the next release I'll consider updating the SDKs examples to show also the optional reauthentication call.

**Comment by archoo** [comment]
ok, thanks.. makes sense.. and i figured that was the case, since it makes sense from a security point of view.. 

i also just realised that since i'm collecting the new password as part of the update request i can just silently call my login() function again as part of that flow and effectively get the outcome i am after..

would definitely be a worthwhile note somewhere in the docs, but hopefully i've described it clearly enough in this post that it helps out the next person :)

cheers!!
